Buildings/Fluid/BuriedPipes/Examples/DiscretizedBuriedPipe.mo

```diff
@@ -0,0 +1,189 @@
+﻿within Buildings.Fluid.BuriedPipes.Examples;
+model DiscretizedBuriedPipe
+  "Example model of a buried pipe with multiple segment"
+  extends Modelica.Icons.Example;
+
+  parameter Modelica.SIunits.Length totLen=10000 "Total pipe length";
+
+  parameter Integer nSeg=2 "Number of pipe segments";
+  parameter Modelica.SIunits.Length segLen[nSeg] = fill(totLen/nSeg,nSeg)
+    "Pipe segments length";
+
+  parameter Integer nSegRev=10 "Number of pipe segments";
+  parameter Modelica.SIunits.Length segLenRev[nSegRev] = fill(totLen/nSegRev,nSegRev)
+    "Pipe segments length";
+
+  parameter Modelica.SIunits.Length  dIns=0.2
+    "Virtual insulation layer thickness";
+
+  replaceable parameter
+    Buildings.BoundaryConditions.GroundTemperature.ClimaticConstants.Boston
+    cliCon "Surface temperature climatic conditions";
+  replaceable parameter Buildings.HeatTransfer.Data.Soil.Generic
+    soiDat(k=1.58,c=1150,d=1600) "Soil thermal properties";
```

Comment:
It would be clearer to have graphical annotations for those declarations so that we can simply access the inner parameters.
This remark holds for all the examples.

Reply:
Did you mean specifically for `soiDat` or also for other objects? I've added graphical annotation for `soiDat` for all examples.

Reply:
I meant for all instances of any record class. We usually instantiate them by drag and drop (so with graphical annotations).